pkg/controller/kubelet-config/helpers.go

```diff
@@ -74,47 +72,41 @@ func validateUserKubeletConfig(cfg *mcfgv1.KubeletConfig) error {
 	if err != nil {
 		return fmt.Errorf("KubeletConfig could not be unmarshalled, err: %v", err)
 	}
-	kcValues := reflect.ValueOf(*kcDecoded)
-	if !kcValues.IsValid() {
-		return fmt.Errorf("KubeletConfig is not valid")
+
+	// Check all the fields a user cannot set within the KubeletConfig CR.
+	// If a user were to set these values, the system may become unrecoverable
+	// (ie: not recover after a reboot).
+	// If the KubeletConfig CR instance contains a non-zero or non-empty value for
+	// one of those fields, the MCC will not apply the CR and log the message.
+
+	// CgroupDriver string `json:"cgroupDriver,omitempty"`
+	if kcDecoded.CgroupDriver != "" {
+		return fmt.Errorf("KubeletConfiguration: CgroupDriver is not allowed to be set")
 	}
-	for _, bannedFieldName := range blacklistKubeletConfigurationFields {
-		v := kcValues.FieldByName(bannedFieldName)
-		if !v.IsValid() {
-			continue
-		}
-		err := fmt.Errorf("%v is not allowed to be set", bannedFieldName)
-		switch v.Kind() {
-		case reflect.Slice:
-			if v.Len() > 0 {
-				return err
-			}
-		case reflect.String:
-			if v.String() != "" {
-				return err
-			}
-		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
-			if v.Int() != 0 {
-				return err
-			}
-		case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
-			if v.Uint() != 0 {
-				return err
-			}
-		case reflect.Struct:
-			if v.Type().String() == "v1.Duration" {
-				d := v.Interface().(metav1.Duration)
-				if d.Duration.String() != "0s" {
-					return err
-				}
-			}
-		case reflect.Map:
-			if len(v.MapKeys()) > 0 {
-				return err
-			}
-		default:
-			return fmt.Errorf("Invalid type in field %v", bannedFieldName)
-		}
+
+	// ClusterDNS []string `json:"clusterDNS,omitempty"`
+	if len(kcDecoded.ClusterDNS) > 0 {
+		return fmt.Errorf("KubeletConfiguration: ClusterDNS is not allowed to be set")
+	}
+
+	// ClusterDomain string `json:"clusterDomain,omitempty"`
+	if kcDecoded.ClusterDomain != "" {
+		return fmt.Errorf("KubeletConfiguration: ClusterDomain is not allowed to be set")
+	}
+
+	// FeatureGates map[string]bool `json:"featureGates,omitempty"`
+	if len(kcDecoded.FeatureGates) > 0 {
+		return fmt.Errorf("KubeletConfiguration: FeatureGates is not allowed to be set")
+	}
+
+	// RuntimeRequestTimeout metav1.Duration `json:"runtimeRequestTimeout,omitempty"`
```

Comment:
can you drop all these comments? 👼

Reply:
sure. thought it would be nice to know what we are operating on here without looking it up. removing